src/overlay/PeerDoor.cpp

```diff
@@ -59,10 +61,17 @@ PeerDoor::acceptNextPeer()
     }
 
     CLOG_DEBUG(Overlay, "PeerDoor acceptNextPeer()");
-    auto sock = make_shared<TCPPeer::SocketType>(mApp.getClock().getIOContext(),
-                                                 TCPPeer::BUFSZ);
+    // Asio guarantees it is safe to create a socket object with overlay's
+    // io_context on main (as long as the socket object isn't shared across
```

Comment:
You use this phrase ("so long as the socket object isn't shared across threads") in another similar comment in TCPPeer and I find it a little confusing: the socket object seems to be "shared" across threads in this codebase, in the sense that a single TCPPeer::SocketType value is periodically accessed from one thread or the other; I think what you're saying is that the object and its methods do not perform any synchronization internally, and so must be protected from simultaneous access from multiple threads by some kind of higher level synchronization. Is that right?

Reply:
That's correct. I re-phrased the comment to clarify this.